src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,148 @@
+    public Todo makeNewTodoTask(String taskString) throws DukeException {
+        String[] todoTokens = taskString.split(" ");
+
+        // If there is a description
+        if (todoTokens.length > 1) {
+            return new Todo(taskString.replaceFirst("^todo ", ""));
+        }
+
+        throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+    }
+
+    /**
+     * Try to make a new event task from the input.
+     *
+     * @param  taskString   the input which contains the event task information
+     * @return  task    the new event task
+     */
+    public Event makeNewEventTask(String taskString) throws DukeException {
+        String[] eventTokens = taskString.split(" /at ");
+
+        // If there is a description and a time/date
+        if (eventTokens.length > 1) {
+            String dateOrTime = eventTokens[1];
+            String[] dateOrTimeTokens = dateOrTime.split(" ");
+
+            // If there is a date and time
+            if (dateOrTimeTokens.length <= 1) {
+                throw new DukeException("☹ OOPS!!! duke.task.Event tasks require a specific time and date.");
+            }
+
+            try {
+                LocalDate date = LocalDate.parse(dateOrTimeTokens[0]);
+                String description = eventTokens[0].replaceFirst("^event ", "");
+                return new Event(description, date, dateOrTimeTokens[1]);
+            } catch (Exception e) {
+                throw new DukeException("☹ OOPS!!! Cannot parse date or time.");
+            }
+        }
+
+        throw new DukeException("☹ OOPS!!! duke.task.Event tasks require a description, "
+                + "and a specific time and date (e.g. 2019-12-12 2-4pm).");
+    }
+
+    /**
+     * Try to make a new deadline task from the input.
+     *
+     * @param  taskString   the input which contains the deadline task information
+     * @return  task    the new deadline task
+     */
+    public Deadline makeNewDeadlineTask(String taskString) throws DukeException {
+        String[] deadlineTokens = taskString.split(" /by ");
+
+        // If there is a description and a deadline
+        if (deadlineTokens.length > 1) {
+            String dateOrTime = deadlineTokens[1];
+            String[] dateOrTimeTokens = dateOrTime.split(" ");
+
+            // If there is a date and time
+            if (dateOrTimeTokens.length <= 1) {
+                throw new DukeException("☹ OOPS!!! duke.task.Deadline tasks require a specific time and date.");
+            }
+
+            try {
+                LocalDate date = LocalDate.parse(dateOrTimeTokens[0]);
+                LocalTime time = LocalTime.parse(dateOrTimeTokens[1], DateTimeFormatter.ofPattern("HHmm"));
+                String description = deadlineTokens[0].replaceFirst("^deadline ", "");
+                return new Deadline(description, date, time);
+            } catch (Exception e) {
+                throw new DukeException("☹ OOPS!!! Cannot parse date or time.");
```

Comment:
i think the error message here does not indicate to the user about what is the correct format of date and time.